lib/libutils/isoc/bget_malloc.c

```diff
@@ -795,9 +821,13 @@ void *malloc(size_t size)
 
 static void free_helper(void *ptr, bool wipe)
 {
-	uint32_t exceptions = malloc_lock(&malloc_ctx);
+	uint32_t exceptions = 0;
+
+	if (wipe)
+		raw_wipe(ptr, &malloc_ctx);
```

Comment:
You're accessing the internal data structure of the heap without any lock, this is bound to cause problems.